pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -31,9 +31,37 @@ import (
 	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/apimachinery/registered"
 	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
+type thirdPartyObjectConverter struct {
+	converter runtime.ObjectConvertor
+}
+
+func (t *thirdPartyObjectConverter) ConvertToVersion(in runtime.Object, outVersion string) (out runtime.Object, err error) {
+	switch in.(type) {
+	// This seems weird, but in this case the ThirdPartyResourceData is really just a wrapper on the raw 3rd party data.
+	// The actual thing printed/sent to server is the actual raw third party resource data, which only has one version.
```

Comment:
No, this is different (sorry this is sort of meta level so the explanation is tricky)

There is only one version of the user object (the raw object that is wrapped by the ThirdPartyResourceData) not the ThirdPartyResourceData object (which might be v1beta1 or v1) and the user object can only have one version.  (since user version conversions aren't [currently] supported)